### PR TITLE
Feat/jest-serializer

### PR DESCRIPTION
**Pull request by nikhedonia**
Implements jest-serializes that strips RNW classNames and serializes computed styles.
Furthermore it strips paths from svg tags as they are usually only visually assessable.

### CODE REVIEW

File: packages/jest-serializer/__tests__/serializer.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      }
+    });
+
+    const component = <Text style={[colored, padded]} />;
+    const tree = renderer.create(component);
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("removes paths d props", () => {
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/serializer.test.js]: can this name be more explicit?

File: packages/jest-serializer/__tests__/serializer.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import PropTypes from "prop-types";
+import React from "react";
+import renderer from "react-test-renderer";
+import Enzyme, { shallow } from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("jest serializer", () => {
+  it("removes rnw-classnames", () => {
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/serializer.test.js]: all test names should begin with `should`.... so
> "should emove RNW-generated classNames"
would be better here... in the process of implementing this across all packages

File: packages/jest-serializer/__tests__/serializer.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { Path } from "svgs";
+import PropTypes from "prop-types";
+import React from "react";
+import renderer from "react-test-renderer";
+import Enzyme, { shallow } from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("jest serializer", () => {
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/serializer.test.js]: describes should always start with a capital letter

File: packages/jest-serializer/__tests__/serializer.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("removes paths d props", () => {
+    const component = <Path d="1 2 3 4 5 6" />;
+    const tree = renderer.create(component);
+    expect(tree).toMatchSnapshot();
+  });
+
+  it.only("transforms render props", () => {
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/serializer.test.js]: remove the `only`

File: packages/jest-serializer/package.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  "dev": "src/jest-serializer",
+  "scripts": {
+    "fmt": "prettier --write '**/*.*'",
+    "prettier:diff": "prettier --list-different '**/*.*'",
+    "lint": "eslint . && npm run prettier:diff",
+    "test": "jest --coverage",
+    "dextrose-stories": "dextrose generate-stories $(pwd)",
+    "dextrose-clean": "dextrose clean-stories $(pwd)",
+    "prepublishOnly": "npm run transpile",
+    "transpile": "babel src -d dist"
```
Comment (colincclark) [inline packages/jest-serializer/package.json]: this is missing the `cleanup-dist` script recently added (see other packages in master or cli)

File: packages/jest-serializer/src/jest-serializer.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { StyleSheet } from "react-native-web";
+import omit from "lodash.omit";
+
+function withoutProps(node) {
+  return { ...node, props: {} };
+}
+
+function getType(node) {
+  if (node.type instanceof Function) {
```
Comment (colincclark) [inline packages/jest-serializer/src/jest-serializer.js]: please use destruct

File: packages/jest-serializer/src/jest-serializer.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+function cleanSvgProps(node, props) {
+  const type = getType(node);
+  return type === "path" || type === "polygon" || type === "svg"
+    ? omit(props, ["d", "path", "points", "viewBox"])
+    : props;
+}
+
+function cleanClassNames(names) {
```
Comment (colincclark) [inline packages/jest-serializer/src/jest-serializer.js]: any reason _not_ to use `=>` functions? optional

File: packages/related-articles/__tests__/web/__snapshots__/related-articles.test.js.snap [hunk 8]
```diff
@@ -87,16 +75,11 @@ exports[`Related articles tests on web:  Lead and two template should render one
                   </div>
                 </div>
                 <div
-                  className="leadContentContainerClass rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-p1pxzi rn-minHeight-ifefl9 rn-minWidth-3e7lpq rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
+                  className="leadContentContainerClass"
```
Comment (colincclark) [inline packages/related-articles/__tests__/web/__snapshots__/related-articles.test.js.snap]: 🍆

File: packages/jest-serializer/src/jest-serializer.js (hunk 7 above)
Comment (nikhedonia) [reply packages/jest-serializer/src/jest-serializer.js]: no but I personally prefer to have functions in the global scope.
Furthermore functions have a name field which improves sometimes the stacktrace.

File: packages/jest-serializer/src/jest-serializer.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  const className = (names || "")
+    .replace(/rn-[^-]+-[^- ]+/g, "")
```
Comment (BernieSumption) [inline packages/jest-serializer/src/jest-serializer.js]: `[^-]+` will over-match if react native web ever emits class names with only one dash in them. Suggest rewriting `/rn-\S+/`, and/or adding a test that the cleanClassNames does not remove semantic class names e.g. "rn-xyz accent-colour".

Comment (nikhedonia) [reply packages/jest-serializer/src/jest-serializer.js]: extended test and regex: `[^-^ ]`

**Pull request merged**